### PR TITLE
Don't allow send empty URL

### PR DESCRIPTION
**Pull request by vitorsouzaalmeida**
emptyUrl

### CODE REVIEW

File: src/components/EditPhoto.jsx [hunk 1]
```diff
@@ -28,10 +28,14 @@ export const EditPhoto = ({ handleUpdate, showInput, shouldShowInput }) => {
       <Button
         variant="contained"
         color="success"
-        style={{margin:"10px"}}
+        style={{ margin: "10px" }}
         onClick={() => {
-          handleUpdate(imageUrl);
-          setImageUrl("");
+          if (imageUrl == undefined) {
```
Comment (NicolasLopes7) [inline src/components/EditPhoto.jsx:33]: we can try to do a smarter validation, verifying if what the user is inputting is properly a link

Comment (NicolasLopes7) [reply src/components/EditPhoto.jsx:33]:
```suggestion
          if (imageUrl.match(/^(?:(?:https?|ftp):\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,}))\.?)(?::\d{2,5})?(?:[/?#]\S*)?$/i)) {
```

File: src/components/AddPhoto.jsx [hunk 2]
```diff
@@ -14,12 +15,32 @@ function AddPhoto({ refetchPhotos }) {
...
+      <GitHub /> <br />
+      <br />
+      <Button
+        variant="contained"
+        color="success"
+        onClick={() => {
+          if (new URL(url)) {
+            handleSubmit();
+          } else {
+            throw new Error("Invalid URL");
```
Comment (fersilva16) [inline src/components/AddPhoto.jsx:37]: URL throws an error when the url is invalid, you should handle that behavior properly with try/catch, create a new function for that

**Review by fersilva16 — APPROVED** [review]

File: src/components/EditPhoto.jsx [hunk 3]
```diff
@@ -28,10 +28,14 @@ export const EditPhoto = ({ handleUpdate, showInput, shouldShowInput }) => {
...
-        style={{margin:"10px"}}
+        style={{ margin: "10px" }}
         onClick={() => {
-          handleUpdate(imageUrl);
-          setImageUrl("");
+          if (new URL(url)) {
+            handleUpdate(imageUrl);
+            setImageUrl("");
+          } else {
+            throw new Error("Invalid URL");
```
Comment (NicolasLopes7) [inline src/components/EditPhoto.jsx:37]: you should create a state to show what error the user gets and show it below the input. Also is good to encapsulate it on a external function

**Comment by vitorsouzaalmeida** [comment]
I think it's ok now. Probably have something to improve in my code, but the problem is solved.

File: src/components/AddPhoto.jsx [hunk 4]
```diff
@@ -1,25 +1,68 @@
...
+  const handleClose = (event, reason) => {
+    if (reason === "clickaway") {
+      return;
+    }
+
+    setOpen(false);
+  };
+
+  const verifyIfImageUrlIsValid = () => {
+    try {
```
Comment (daniloab) [inline src/components/AddPhoto.jsx:28]: you can resolve this with yup validation

**Pull request merged**